internal/mode/static/nginx/runtime/verify_test.go

```diff
@@ -68,7 +89,88 @@ func TestVerifyClient(t *testing.T) {
 		t.Run(test.name, func(t *testing.T) {
 			g := NewWithT(t)
 
-			err := c.waitForCorrectVersion(test.ctx, test.expectedVersion)
+			err := c.waitForCorrectVersion(test.ctx, test.expectedVersion, "/childfile", []byte("1 2 3"), test.readFile)
+
+			if test.expectError {
+				g.Expect(err).To(HaveOccurred())
+			} else {
+				g.Expect(err).ToNot(HaveOccurred())
+			}
+		})
+	}
+}
+
+func TestEnsureNewNginxWorkers(t *testing.T) {
+	previousContents := []byte("1 2 3")
+	newContents := []byte("4 5 6")
+
+	readFileError := func(string) ([]byte, error) {
+		return nil, errors.New("error")
+	}
+
+	readFilePrevious := func(string) ([]byte, error) {
+		return previousContents, nil
+	}
+
+	readFileNew := func(string) ([]byte, error) {
+		return newContents, nil
+	}
+
+	ctx := context.Background()
+
+	cancellingCtx, cancel := context.WithCancel(ctx)
+	time.AfterFunc(10000*time.Millisecond, cancel)
+
+	cancellingCtx2, cancel2 := context.WithCancel(ctx)
+	time.AfterFunc(1*time.Millisecond, cancel2)
+
+	tests := []struct {
+		ctx              context.Context
+		readFile         readFileFunc
+		name             string
+		previousContents []byte
+		expectError      bool
+	}{
+		{
+			ctx:              ctx,
+			readFile:         readFileNew,
+			previousContents: previousContents,
+			expectError:      false,
+			name:             "normal case",
+		},
+		{
+			ctx:              ctx,
+			readFile:         readFileError,
+			previousContents: previousContents,
+			expectError:      true,
+			name:             "cannot read file",
+		},
+		{
+			ctx:              cancellingCtx,
+			readFile:         readFilePrevious,
+			previousContents: previousContents,
+			expectError:      true,
+			name:             "no new workers",
```

Comment:
Since this case accepts the context with the longer timeout, are we also testing the retry mechanism? If so, can you add that tidbit to the name? Something like "timed out waiting for new workers"

Reply:
Good call, yes it is testing the retry mechanism where the other one isn't - I'll ass more context